controllers/tektontasks_controller.go

```diff
@@ -175,6 +185,94 @@ func (r *tektonTasksReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+func (r *tektonTasksReconciler) checkDeployedCRs(ctx context.Context, tektonRequest *common.Request) (bool, error) {
+	tektonTasksCRList, err := r.getCRList(ctx)
+	if err != nil {
+		tektonRequest.Logger.Error(err, "CRS listing failed")
+		return true, err
+	}
+
+	if len(tektonTasksCRList.Items) > 1 {
+		err := fmt.Errorf("there are multiple CRs deployed")
+		tektonRequest.Logger.Error(err, err.Error())
+
+		for _, tektonTask := range tektonTasksCRList.Items {
+			if tektonTask.Name != tektonRequest.Name {
+				degraded := false
+				for _, condition := range tektonTask.Status.Conditions {
+					if condition.Type == conditionsv1.ConditionDegraded {
+						degraded = condition.Status == v1.ConditionTrue
+						break
+					}
+				}
+
+				if !degraded {
+					namespacedName := types.NamespacedName{
+						Namespace: tektonTask.Namespace,
+						Name:      tektonTask.Name,
+					}
+					tektonTasks := &tekton.TektonTasks{}
+
+					errGet := r.client.Get(ctx, namespacedName, tektonTasks)
+					if errGet != nil {
+						return true, errGet
+					}
+
+					_, errSetStatus := r.setStatusMultipleCRs(tektonTasks, ctx, err)
+					if errSetStatus != nil {
+						return true, errSetStatus
+					}
+				}
+			}
+		}
+		return false, err
+	}
+	return false, nil
+}
+
+func (r *tektonTasksReconciler) getCRList(ctx context.Context) (*tekton.TektonTasksList, error) {
+	tektonTasksCRList := &tekton.TektonTasksList{}
+	tektonTasksCRList.SetGroupVersionKind(schema.GroupVersionKind{
+		Kind:    "TektonTasks",
+		Version: "tektontasks.kubevirt.io/v1alpha1",
+	})
+
+	err := r.client.List(ctx, tektonTasksCRList)
+	if err != nil {
+		return nil, err
+	}
+
+	return tektonTasksCRList, nil
+}
+
+func (r *tektonTasksReconciler) setStatusMultipleCRs(tektonTasks *tekton.TektonTasks, ctx context.Context, err error) (reconcile.Result, error) {
+	conditionsv1.SetStatusCondition(&tektonTasks.Status.Conditions, conditionsv1.Condition{
+		Type:    conditionsv1.ConditionAvailable,
+		Status:  v1.ConditionFalse,
+		Reason:  "Available",
+		Message: err.Error(),
+	})
+	conditionsv1.SetStatusCondition(&tektonTasks.Status.Conditions, conditionsv1.Condition{
+		Type:    conditionsv1.ConditionProgressing,
+		Status:  v1.ConditionFalse,
+		Reason:  "Progressing",
+		Message: err.Error(),
+	})
+	conditionsv1.SetStatusCondition(&tektonTasks.Status.Conditions, conditionsv1.Condition{
+		Type:    conditionsv1.ConditionDegraded,
+		Status:  v1.ConditionTrue,
+		Reason:  "Degraded",
+		Message: err.Error(),
+	})
+
+	err = r.client.Status().Update(ctx, tektonTasks)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
```

Comment:
I would suggest letting this method only return an `err`. Leave it up to the caller of the method to pass up the `err` or e.g. `return ctrl.Result{}, nil`.